wandb/sdk/launch/environment/aws_environment.py

```diff
@@ -0,0 +1,238 @@
+    "it with `pip install wandb[launch]`.",
+)
+botocore = get_module(
+    "botocore",
+    required="AWS environment requires botocore to be installed. Please install "
+    "it with `pip install wandb[launch]`.",
+)
+
+s3_uri_re = re.compile(r"s3://([^/]+)/(.+)")
+
+_logger = logging.getLogger(__name__)
+
+
+class AwsEnvironment(AbstractEnvironment):
+    """AWS environment."""
+
+    def __init__(
+        self,
+        region: str,
+        access_key: str,
+        secret_key: str,
+        session_token: str,
+        verify: bool = True,
+    ) -> None:
+        """Initialize the AWS environment.
+
+        Args:
+            region (str): The AWS region.
+
+        Raises:
+            LaunchError: If the AWS environment is not configured correctly.
+        """
+        super().__init__()
+        _logger.info(f"Initializing AWS environment in region {region}.")
+        self._region = region
+        self._access_key = access_key
+        self._secret_key = secret_key
+        self._session_token = session_token
+        if verify:
+            self.verify()
+
+    @classmethod
+    def from_default(cls, region: str, verify: bool = True) -> "AwsEnvironment":
+        """Create an AWS environment from the default AWS environment.
+
+        Args:
+            region (str): The AWS region.
+            verify (bool, optional): Whether to verify the AWS environment. Defaults to True.
+
+        Returns:
+            AwsEnvironment: The AWS environment.
+        """
+        _logger.info("Creating AWS environment from default credentials.")
+        try:
+            session = boto3.Session()
+            credentials = session.get_credentials()
+            access_key = credentials.access_key
+            secret_key = credentials.secret_key
+            session_token = credentials.token
+        except botocore.client.ClientError as e:
+            raise LaunchError(
+                f"Could not create AWS environment from default environment. Please verify that your AWS credentials are configured correctly. {e}"
+            )
+        return cls(
+            region=region,
+            access_key=access_key,
+            secret_key=secret_key,
+            session_token=session_token,
+            verify=verify,
+        )
+
+    def verify(self) -> None:
+        """Verify that the AWS environment is configured correctly.
+
+        Raises:
+            LaunchError: If the AWS environment is not configured correctly.
+        """
+        _logger.debug("Verifying AWS environment.")
+        try:
+            session = self.get_session()
+            client = session.client("sts")
+            client.get_caller_identity()
+            # TODO: log identity details from the response
+        except botocore.exceptions.ClientError as e:
+            raise LaunchError(
+                f"Could not verify AWS environment. Please verify that your AWS credentials are configured correctly. {e}"
```

Comment:
Also `raise LaunchError(...) from e` here. Makes the error more clear for the reader